src/daft-parquet/src/stream_reader.rs

```diff
@@ -57,6 +57,7 @@ pub(crate) fn arrow_column_iters_to_table_iter(
     original_columns: Option<Vec<String>>,
     original_num_rows: Option<usize>,
 ) -> impl Iterator<Item = DaftResult<Table>> {
+    assert!(!arr_iters.is_empty());
```

Comment:
Alternatively, return an Option and do a None on an empty input iter so the caller has a chance to catch the empty iter case?

Reply:
That's a great suggestion! much cleaner